apps/api_web/lib/api_web/controllers/route_controller.ex

```diff
@@ -143,6 +143,11 @@ defmodule ApiWeb.RouteController do
     |> Route.select()
   end
 
+  defp do_filter(%{service_ids: []}), do: []
+
+  defp do_filter(%{service_ids: [service_id | _], type: types}),
+    do: service_id |> RoutesByService.for_service_id_and_types(types) |> Route.by_ids()
```

Comment:
Why only filter by the first service ID, and not all service IDs that are active?

Reply:
eh i misread the `do_format_filter` that converts `date` to `service_ids`. I originally thought that it ultimately only returned 1 `service_id`. I'll change this to use all and add a test case.

Reply:
Should be resolved now